### PR TITLE
Introduced exampleblockGB environment for examples with godbolt links

### PR DESCRIPTION
**Pull request by sponce**
And added also an index of godbolt code examples at the end with links both to the slide concerned and to godbolt

### CODE REVIEW

File: talk/setup.tex [hunk 1]
```diff
@@ -189,10 +187,38 @@
...
-  \href{#1}{\color{blue!50!white} godbolt}%
+\newcommand\godboltLink[2]{%
+  \ifnum \overlaynumber=1%
+    \href{#1}{\color{blue!50!white} godbolt}%
+    \hypertarget<1>{godbolt.\thegb@counter}{}
+    \addtocontents{lgb}{\protect\item \protect\hyperlink{godbolt.\thegb@counter}{#2 - \href{#1}{godbolt}\protect\hfill\insertframenumber}}%
+    \stepcounter{gb@counter}%
+  \fi
 }
+
```
Comment (hageboeck) [inline talk/setup.tex:213]:
```suggestion

% Use: \begin{exampleblockGB}{<Box Title>}{<godbolt link>}{<title in index>}}
```

**Review by hageboeck — APPROVED** [review]
👍 
Maybe just add a comment on how to use it as suggested below?

File: talk/setup.tex [hunk 2]
```diff
@@ -189,10 +187,38 @@
...
+\regtotcounter{gb@counter}
+\resetcounteronoverlays{gb@counter}
+\newcommand*{\overlaynumber}{\number\beamer@slideinframe}
+
 % Use as \godbolLink{url}
-\newcommand\godboltLink[1]{%
-  \href{#1}{\color{blue!50!white} godbolt}%
+\newcommand\godboltLink[2]{%
+  \ifnum \overlaynumber=1%
+    \href{#1}{\color{blue!50!white} godbolt}%
```
Comment (hageboeck) [inline talk/setup.tex:207]:
```suggestion
  \href{#1}{\color{blue!50!white} godbolt}%
    \ifnum \overlaynumber=1%
```
Like this, the link will also show on the other slides, but the entry to the index only happens once.
However, if the box only appears on overlay != 1, there won't be an entry at all. Is this an issue?

Comment (sponce) [reply talk/setup.tex:207]: Oh indeed ! Thanks for spotting.

**Comment by hageboeck** [comment]
This seems to work now. I might copy the technique of inserting only on overlay number 1 for the list of good practices ...

**Pull request merged**